File: BraintreeModels/src/test/java/com/braintreepayments/api/AuthorizationUnitTest.java
package com.braintreepayments.api;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class AuthorizationUnitTest {

    @Test
    public void fromString_returnsValidClientTokenWhenBase64() {
        Authorization authorization = Authorization.fromString(Fixtures.BASE64_CLIENT_TOKEN);

        assertTrue(authorization instanceof ClientToken);
    }

    @Test
    public void fromString_returnsValidTokenizationKey() {
        Authorization authorization = Authorization.fromString(Fixtures.TOKENIZATION_KEY);

        assertTrue(authorization instanceof TokenizationKey);
    }

    @Test
    public void fromString_returnsValidPayPalUAT() {
        Authorization authorization = Authorization.fromString(Fixtures.BASE64_PAYPAL_UAT);

        assertTrue(authorization instanceof PayPalUAT);
    }

    @Test(expected = InvalidArgumentException.class)
    public void fromString_throwsWhenPassedNull() {
        Authorization.fromString(null);
    }

    @Test(expected = InvalidArgumentException.class)
    public void fromString_throwsWhenPassedAnEmptyString() {
        Authorization.fromString("");
    }

    @Test(expected = InvalidArgumentException.class)
    public void fromString_throwsWhenPassedJunk() {
        Authorization.fromString("not authorization");
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/AuthorizationUnitTest.kt
package com.braintreepayments.api.core

import com.braintreepayments.api.core.Authorization.Companion.fromString
import com.braintreepayments.api.testutils.Fixtures
import org.robolectric.RobolectricTestRunner
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class AuthorizationUnitTest {
    @Test
    fun `when given a base64 client token, fromString returns a ClientToken`() {
        val authorization = fromString(Fixtures.BASE64_CLIENT_TOKEN)
        assertTrue(authorization is ClientToken)
    }

    @Test
    fun `when given a base64 client token with spaces, fromString returns a ClientToken`() {
        val authorization = fromString(Fixtures.BASE64_CLIENT_TOKEN_WITH_SPACES)
        assertTrue(authorization is ClientToken)
    }

    @Test
    fun `when given a tokenization key, fromString returns a TokenizationKey`() {
        val authorization = fromString(Fixtures.TOKENIZATION_KEY)
        assertTrue(authorization is TokenizationKey)
    }

    @Test
    fun `when given a tokenization key with spaces, fromString returns a TokenizationKey`() {
        val authorization = fromString(Fixtures.TOKENIZATION_KEY_WITH_SPACES)
        assertTrue(authorization is TokenizationKey)
    }

    @Test
    fun `when passed null, fromString returns InvalidAuthorization`() {
        val result = fromString(null)
        assertTrue(result is InvalidAuthorization)
    }

    @Test
    fun `when passed an empty string, fromString returns InvalidAuthorization`() {
        val result = fromString("")
        assertTrue(result is InvalidAuthorization)
    }

    @Test
    fun `when passed a junk string, fromString returns InvalidAuthorization`() {
        val result = fromString("not authorization")
        assertTrue(result is InvalidAuthorization)
    }
}
